test/test_progress_bar.py

```diff
@@ -0,0 +1,57 @@
+import contextlib
+import io
+
+import diffrax
+import numpy as np
+
+
+def test_tqdm_progress_bar():
+    term = diffrax.ODETerm(lambda t, y, args: -0.2 * y)
+    solver = diffrax.Dopri5()
+    t0 = 0
+    t1 = 5
+    dt0 = 0.01
+    y0 = 1.0
+
+    saveat = diffrax.SaveAt(steps=True)
+    diffrax.diffeqsolve(
+        term,
+        solver,
+        t0,
+        t1,
+        dt0,
+        y0,
+        saveat=saveat,
+        progress_bar=diffrax.TqdmProgressBar(),
+    )
+
+
+def test_text_progress_bar():
+    term = diffrax.ODETerm(lambda t, y, args: -0.2 * y)
+    solver = diffrax.Dopri5()
+    t0 = 0
+    t1 = 5
+    dt0 = 0.01
+    y0 = 1.0
+    saveat = diffrax.SaveAt(steps=True)
+
+    f = io.StringIO()
+    with contextlib.redirect_stdout(f):
```

Comment:
Oh by the way, pytest has a built-in for this: see [here](https://github.com/patrick-kidger/equinox/blob/34252989c251c315c6eee5895dd64e9645b350ce/tests/test_while_loop.py#L196-L198) for an example